tests/components/glances/test_sensor.py

```diff
@@ -43,6 +43,16 @@ async def test_sensor_states(hass: HomeAssistant) -> None:
         assert state.state == HA_SENSOR_DATA["raid"]["md1"]["available"]
     if state := hass.states.get("sensor.0_0_0_0_md1_used"):
         assert state.state == HA_SENSOR_DATA["raid"]["md1"]["used"]
+    test_data_gpu = HA_SENSOR_DATA["gpu"]["GPU_0__NVIDIA GeForce RTX 3080"]
+    sensor_name_gpu = "sensor.0_0_0_0_gpu_0_nvidia_geforce_rtx_3080"
+    if state := hass.states.get(sensor_name_gpu + "_memory_use"):
+        assert state.state == str(test_data_gpu["mem"])
+    if state := hass.states.get(sensor_name_gpu + "_processor_use"):
+        assert state.state == str(test_data_gpu["proc"])
+    if state := hass.states.get(sensor_name_gpu + "_temperature"):
+        assert state.state == str(test_data_gpu["temperature"])
+    if state := hass.states.get(sensor_name_gpu + "_fan_speed"):
+        assert state.state == str(test_data_gpu["fan_speed"])
```

Comment:
@engrbm87 I am a bit surprised that the GPU memory is registered as a string in Home Assistant state, while the sensor for RAM memory use looks very similar and does not need the string cast in the test here. Any idea what's causing this?

Reply:
I can see that the GPU sensors are mostly percentages. For the memory sensors there is actual memory and memory usage as percentage.
[This link](https://github.com/nicolargo/glances/blob/develop/glances/plugins/gpu/__init__.py) shows the UOM for the GPU sensors, please double check that they are the same in our code here.
As for the string issue I am not sure, what error are you getting in the test if you don't cast as str.

Reply:
The HA `state` for the GPU sensors is a string, even though the input data (`HA_SENSOR_DATA` in glances_api) is int or float. Not sure why the sensor state becomes string, as the sensors above in this test seem to have the same sensor implementation but do have a *numerical* HA state (docker memory use for example).

Reply:
I did some digging and found that the sensors tests added in PR #93542 is not using the correct `entity_id`. Because it is using an if statement to get the state variable the assertion is not getting executed which is why we are not getting an assertion error. I think the tests should be fixed in a preliminary PR.
As for the `str` casting, I think this is required because the entity state will always be a string.